presto-main/src/test/java/com/facebook/presto/type/TestRealOperators.java

```diff
@@ -245,4 +248,13 @@ public void testIndeterminate()
         assertOperator(INDETERMINATE, "cast(1.2 as real)", BOOLEAN, false);
         assertOperator(INDETERMINATE, "cast(123 as real)", BOOLEAN, false);
     }
+
+    @Test
+    public void testNanHash()
+    {
+        int nanValues[] = new int[] {floatToIntBits(Float.NaN), floatToRawIntBits(Float.NaN), -0x400000, 0x7fc00000};
```

Comment:
- replace `-0x400000` with normal hex repr, `0xffc00000`
- `0x7fc00000 == floatToIntBits(Float.NaN) == floatToRawIntBits(Float.NaN)`, so its thrice on the list (i still see a value in having explicit hex and what `Float.NaN` yields). Perhaps you want something like 0x7fc01234
- `new int[]` is redundant

All in all, you could have something like this
```
int nanRepresentations[] = {floatToIntBits(Float.NaN), 0xffc00000, 0x7fc00000, 0x7fc01234, 0xffc01234};
for (int nanRepresentation : nanRepresentations) {
    assertTrue(isNaN(Float.intBitsToFloat(nanRepresentation)));
    assertEquals(RealOperators.hashCode(nanRepresentations[0]), RealOperators.hashCode(nanRepresentation));
}
```